### ISSUE TITLE
Include "git notes" in blame/commit message

### ISSUE THREAD

**Issue by devinrhode2** [gitkraken/vscode-gitlens#2432]
Git has a very interesting feature simply called "git notes".

Simply put, it allows you to arbitrarily attach extra text-based notes to a commit sha, without changing that sha. It stores this data in `refs/notes/commits`. Here is an example `git log` which I added notes to via `git notes add <sha>`:
```
$ git log -1
commit 614fc95005b9983d6d4f1d67e85949a0998be9c2 (HEAD -> main)
Author: devinrhode2 <devinrhode2@gmail.com>
Date:   Fri Dec 30 15:20:06 2022 -0500

    Tweak .gitconfig comment

Notes:
    This is a sample message written inside a .git/NOTES_EDITMSG text buffer/prompt.
    
    I ran the command `git notes add` with no params.
```

Currently gitlens does not show anything from this `Notes:` section, which IS displayed via normal `git log` commands:
![CleanShot 2023-01-03 at 13 09 45](https://user-images.githubusercontent.com/539816/210419182-dfc7a8aa-138b-426b-bba0-8a8dcae285ba.png)

Supporting `git notes` is probably an relatively simple task, only requiring a slight tweak to some `git log` command inside gitlens.

Some interesting applications of `git notes`:
 - Github.com of course supports commenting on specific commits, those comments are stored in Github's internal DB. IMO it may make more sense for github to store general comments inside `git notes` i.e. `refs/notes/commits`. Git currently does not support per-line comments, but there could be an agreed upon format for storing data inside git refs, which gitlens can then interpret, ultimately enabling per-line/sha comments/notes.
 - Another interesting use for git notes: Add diffs for test/type-checking/linter output. For example, a command could be written which runs typescript on a javascript codebase (temporarily add a tsconfig.json, run tsc, store output, go to next sha, diff output of tsc, etc).
 - Same thing could be done with output from a test runner like jest. This can help answer the question of "When did this test warning start appearing?".

Git also has various options for controlling which "refs" are displayed in which contexts. For example, there could be `refs/notes/commits/tsc-diffs` and `.../jest-diffs` and `.../eslint-diffs`, you can then play around with `core.notesRef` and/or `notes.displayRef` to allow just showing tsc-diffs or jest-diffs, depending on what you are focusing on at the time.

**Comment by d13** [comment]
There's several really neat ideas to think about here, thanks for posting!

**Comment by devinrhode2** [comment]
Yeah I think Nx dev tools/Narwhal might be really interested in automating some commit history.

We generally know which commits have a passing build, or which commit introduced some warning in the test suite, _when we author and land_ those commits, but that information isn't systematically recorded.

Letting errors/warnings pile up can be really painful. Eventually a new dev comes in, and has no idea if they have added a new one or not. Furthermore, if a warning or error seems like it ought to be fixed, it feels mostly impossible to track down which code introduced that error/warning. Git notes/commit messages can help us keep track of when these warnings/errors were introduced. Even plain text files of the errors/warnings, so you get a normal git diff.

Having history be "indented" by each error/warning can also help us visualize technical debt.

**Comment by Mattwmaster58** [comment]
This would enable a workflow for me that's pretty niche but super useful so I'd fully support. 

I'm definitely the minority though with this specific workflow, but I'm sure there's others (dozens of us) that would benefit in ways we aren't aware of, but I think it's cool to enable people to do more stuff with the extension. For those that can't wait, you can just hack this in _(really hacky)._ It probably will break some things, but nothing I care about yet:

```pwsh
$filePath = "C:\Users\<your username>\.vscode\extensions\eamodio.gitlens-<your version>\dist\gitlens.js"
(Get-Content $filePath -Raw) -replace '%n%B%n', '%n%B%n%N%n' | Set-Content -Path $filePath -NoNewline
```

To properly support this we should probably surround it in tags, parse it out, and display it with some distinction showing that it's not part of the message. What the replacement does above is pretend it's part of the commit message.
